Spectral-cube uses astropy.coordinates.angle_utilities which is removed in Astropy 6

With Astropy 6.0, the CI tests of spectral-cube no longer pass; there are now many failures like
```python
______________________ test_casa_read_basic[False-False] _______________________

memmap = False, bigendian = False

    @pytest.mark.parametrize(('memmap', 'bigendian'), product((False, True), (False, True)))
    def test_casa_read_basic(memmap, bigendian):
        […]
>       cube.moment0()

/usr/lib/python3/dist-packages/spectral_cube/tests/test_casafuncs.py:133: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3/dist-packages/spectral_cube/spectral_cube.py:1702: in moment0
    return self.moment(axis=axis, order=0, how=how)
/usr/lib/python3/dist-packages/spectral_cube/dask_spectral_cube.py:1069: in moment
    pix_cen = self._pix_cen()[axis]
/usr/lib/python3/dist-packages/spectral_cube/utils.py:18: in wrapper
    self._cache[(func, args)] = func(self, *args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = DaskSpectralCube with shape=(3, 4, 5) and unit=Jy / beam and chunk size (3, 4, 5):
 n_x:      5  type_x: RA---SIN  uni...9.934094 deg:   29.935778 deg
 n_s:      3  type_s: FREQ      unit_s: Hz     range: 1421917021.146 Hz:1421929254.325 Hz

    @cached
    def _pix_cen(self):
        """…"""
        # Start off by extracting the world coordinates of the pixels
        _, lat, lon = self.world[0, :, :]
        spectral, _, _ = self.world[:, 0, 0]
        spectral -= spectral[0] # offset from first pixel
    
        # Convert to radians
        lon = np.radians(lon)
        lat = np.radians(lat)
    
        # Find the dx and dy arrays
>       from astropy.coordinates.angle_utilities import angular_separation
E       ModuleNotFoundError: No module named 'astropy.coordinates.angle_utilities'

/usr/lib/python3/dist-packages/spectral_cube/spectral_cube.py:1456: ModuleNotFoundError
```
[full test log](https://ci.debian.net/data/autopkgtest/unstable/amd64/s/spectral-cube/40223535/log.gz)

Reason is that **angle_utilities** was deprecated in Astropy 4.2 and [removed in 6.0](https://github.com/astropy/astropy/blob/main/CHANGES.rst#astropycoordinates-1):
> Removed deprecated angle parsing and formatting utilities from angle_utilities. Use the functions from angle_formats instead. [#14675]

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -42,7 +42,7 @@ jobs:
           - os: ubuntu-latest
             python-version: 3.9
             name: Py3.9 all
-            toxenv: py39-test-viz-noviz-docs
+            toxenv: py39-test-viz-noviz
           - os: ubuntu-latest
             python-version: 3.9
             name: Py3.9 dev
@@ -58,7 +58,7 @@ jobs:
           - os: ubuntu-latest
             python-version: 3.9
             name: Py3.9 all dev
-            toxenv: py39-test-viz-noviz-docs-dev
+            toxenv: py39-test-viz-noviz-dev
           - os: ubuntu-latest
             python-version: 3.8
             name: Py3.8 mindeps and CASA

--- a/.readthedocs.yml
+++ b/.readthedocs.yml
@@ -1,11 +1,12 @@
 version: 2
 
 build:
-  image: latest
+  os: "ubuntu-20.04"
+  tools:
+    python: "mambaforge-4.10"
 
 # Install regular dependencies.
 python:
-  version: 3.7
   install:
     - method: pip
       path: .

--- a/.readthedocs.yml
+++ b/.readthedocs.yml
@@ -3,7 +3,7 @@ version: 2
 build:
   os: "ubuntu-20.04"
   tools:
-    python: "mambaforge-4.10"
+    python: "3.12"
 
 # Install regular dependencies.
 python:

--- a/spectral_cube/spectral_cube.py
+++ b/spectral_cube/spectral_cube.py
@@ -1453,7 +1453,7 @@ class BaseSpectralCube(BaseNDClass, MaskableArrayMixinClass,
         lat = np.radians(lat)
 
         # Find the dx and dy arrays
-        from astropy.coordinates.angle_utilities import angular_separation
+        from astropy.coordinates import angular_separation
         dx = angular_separation(lon[:, :-1], lat[:, :-1],
                                 lon[:, 1:], lat[:, :-1])
         dy = angular_separation(lon[:-1, :], lat[:-1, :],
@@ -1541,7 +1541,7 @@ class BaseSpectralCube(BaseNDClass, MaskableArrayMixinClass,
         lat = np.radians(lat)
 
         # Find the dx and dy arrays
-        from astropy.coordinates.angle_utilities import angular_separation
+        from astropy.coordinates import angular_separation
         dx = angular_separation(lon[:, :-1], lat[:, :-1],
                                 lon[:, 1:], lat[:, :-1])
 
@@ -1561,7 +1561,7 @@ class BaseSpectralCube(BaseNDClass, MaskableArrayMixinClass,
         lat = np.radians(lat)
 
         # Find the dx and dy arrays
-        from astropy.coordinates.angle_utilities import angular_separation
+        from astropy.coordinates import angular_separation
         dy = angular_separation(lon[:-1, :], lat[:-1, :],
                                 lon[1:, :], lat[1:, :])
 

